tests/cases/dom/events/mediaQuery.js

```diff
@@ -0,0 +1,195 @@
+import {mediaQuery} from "../../../../dom/events";
+import QUnit from "qunit";
+import {findOne} from "../../../../dom/traverse";
+import {createElement} from "../../../../dom/manipulate";
+import {setStyles} from "../../../../dom/css";
+
+const matchQueryList = [];
+
+QUnit.module("dom/events/mediaQuery()", {
+    beforeEach: () => {
+        findOne("#qunit-fixture").innerHTML = `
+            <div class="example"></div>
+        `;
+    },
+    afterEach: () => {
+        matchQueryList.forEach((element, index) => {
+            if (undefined !== element.mediaQuery)
+            {
+                if (undefined !== element.mediaQuery.remove)
+                {
+                    element.remove();
+                }
+
+                if (undefined !== element.mediaQuery.targetWindow)
+                {
+                    element.targetWindow.close();
+                }
+            }
+
+            matchQueryList.splice(index, 1);
+        });
+    }
+});
+
+
+QUnit.test(
+    "mediaQuery() on window media query matches without immediate execution",
+    (assert) =>
+    {
+        assert.expect(0);
+
+        const callback = (matches) => {
+            assert.ok(matches, "Media query matches.");
+        };
+
+        matchQueryList.push(mediaQuery(`(min-width: ${window.innerWidth}px)`, callback));
+    }
+);
+
+
+QUnit.test(
+    "mediaQuery() on window media query matches with immediate execution",
+    (assert) =>
+    {
+        const done = assert.async(1);
+
+        assert.expect(1);
+
+        const callback = (matches) => {
+            assert.ok(matches, "Media query matches.");
+            done();
+        };
+
+        matchQueryList.push(mediaQuery(`(min-width: ${window.innerWidth}px)`, callback, true));
+    }
+);
+
+
+QUnit.test(
+    "mediaQuery() on window media query not matching without immediate execution",
+    (assert) =>
+    {
+        assert.expect(0);
+
+        const callback = (matches) => {
+            assert.notOk(matches, "Media query does not match.");
+        };
+
+        matchQueryList.push(mediaQuery(`(min-width: ${window.innerWidth + 1}px)`, callback));
+    }
+);
+
+
+QUnit.test(
+    "mediaQuery() on window media query not matching with immediate execution",
+    (assert) =>
+    {
+        const done = assert.async(1);
+
+        assert.expect(1);
+
+        const callback = (matches) => {
+            assert.notOk(matches, "Media query does not match.");
+            done();
+        };
+
+        matchQueryList.push(mediaQuery(`(min-width: ${window.innerWidth + 1}px)`, callback, true));
+    }
+);
+
+QUnit.test(
+    "mediaQuery() on window media query matching without immediate execution and window size changing",
```

Comment:
We don't need to test whether `matchMedia` works. The browsers are supposed to test that.

You should however mock the `matchMedia()` call on the window and check that it is properly called.